app/services/finance/npq/course_statement_calculator.rb

```diff
@@ -108,7 +109,32 @@ def monthly_service_fees
       end
 
       def course_total
-        monthly_service_fees + output_payment_subtotal - clawback_payment
+        monthly_service_fees + output_payment_subtotal - clawback_payment + targeted_delivery_funding_subtotal
+      end
+
+      def course_has_targeted_delivery_funding?
+        show_targeted_delivery_funding? &&
+          !(::Finance::Schedule::NPQEhco::IDENTIFIERS + ::Finance::Schedule::NPQSupport::IDENTIFIERS).compact.include?(course.identifier)
+      end
+
+      def targeted_delivery_funding_declarations_count
+        return 0 unless course_has_targeted_delivery_funding?
+
+        pp_ids = statement
+          .billable_statement_line_items
+          .joins(:participant_declaration)
+          .where(participant_declarations: { course_identifier: course.identifier })
+          .map { |l| l.participant_declaration.participant_profile_id }
+
+        NPQApplication.where(id: pp_ids, targeted_delivery_funding_eligibility: true, eligible_for_funding: true).count
```

Comment:
Played around with this, maybe we can try this query instead, as I think we need to make sure we get unique declarations per user per course identifier as a participant can do multiple NPQ courses:
```
statement
  .billable_statement_line_items
  .joins(:participant_declaration)
  .joins("INNER JOIN npq_applications  ON npq_applications.id = participant_declarations.participant_profile_id")
  .where(
    participant_declarations: { course_identifier: course.identifier }, 
    npq_applications: {targeted_delivery_funding_eligibility: true, eligible_for_funding: true}
  )
  .merge(ParticipantDeclaration.select("DISTINCT (user_id, declaration_type)"))
  .count
```

Reply:
could we memoize this as well?

Reply:
Replaced the code with this